Makefile

```diff
@@ -61,7 +64,7 @@ wasp/boards/$(BOARD_SAFE)/watch.py : wasp/boards/$(BOARD_SAFE)/watch.py.in
 micropython/mpy-cross/mpy-cross:
 	$(MAKE) -C micropython/mpy-cross
 
-micropython: build-$(BOARD_SAFE) wasp/boards/$(BOARD_SAFE)/watch.py micropython/mpy-cross/mpy-cross
+micropython: build-$(BOARD_SAFE) wasp/boards/manifest_user_apps.py wasp/boards/$(BOARD_SAFE)/watch.py micropython/mpy-cross/mpy-cross
```

Comment:
I think this new dependency is missing from the simulator (`make sim` currently doesn't work until after you have done a build for the watch).

Reply:
Great catch. Fixed. Thanks for the thorough review. This PR ended up being about twice as big as I was planning haha